### ISSUE TITLE
Missing documentation on aof-use-rdb-preamble

### ISSUE THREAD

**Issue by dvirsky** [antirez/redis-doc#908]
The new persistence mode is not documented outside of redis.conf.

**Comment by antirez** [comment]
I've to pause a bit this new features craziness and document the damn things we already did... Otherwise Redis is going to be like the software I hate, where the documentation is among Stack Overflow or mailing list posts in the hope of people who knows replying and so forth. Thanks for the heads up.

**Comment by dvirsky** [comment]
I'll try to find some time and document this specific thing. 

I can also try to regenerate the documentation for the modules API, which is really out of sync, there should be your ruby script that automates this, right?

**Comment by antirez** [comment]
Thanks, having this could help, but we are now facing a fundamental problem of the whole documentation no longer reflecting the new semantics of Redis in different areas. I need to stop and spend some time on rewriting many parts... There is no need to continue pushing new things forever after all :-D

**Comment by dvirsky** [comment]
You're right - beyond documenting the specific feature wrt aof/rdb - I think the whole document on persistence needs a big rewrite with different emphasis - if you want to point people in that direction.

**Comment by tamalsaha** [comment]
This is an old issue but the documentation is still missing. Any plans to update it? Is there any other place where we can find relevant information for aof-use-rdb-preamble feature?

Some of the questions:

- Does redis rewrite the rdb file when `BGREWRITEAOF` command is issued?
- If this flag is enabled, could we just use the aof presistence option where the aof rewrites produce small files like snapshotting?

**Comment by tomwganem** [comment]
We got bit in the ass with this option being turned on by default in redis 6. We were so confused as to why our normal restore procedure was no longer working, which is all about issuing a "BGREWRITEAOF"  command and piping in the resulting aof via redis-cli.  Of course, that doesn't work with this option turned on.

Why is this suddenly switched to being on in redis 6 without proper documentation? I can readily accept breaking behavior in major version upgrades...when I know about them.

**Comment by mich-elle-luna** [comment]
The persistence doc (https://redis.io/docs/management/persistence/) was update for v7 at least, but still doesn't contain info about aof-use-rbd-preamble that I could find. This is from the recent Redis.conf:

# Redis can create append-only base files in either RDB or AOF formats. Using
# the RDB format is always faster and more efficient, and disabling it is only
# supported for backward compatibility purposes.
aof-use-rdb-preamble yes
